plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/utils/AndroidUtils.java

```diff
@@ -116,9 +117,15 @@ public static List<GradleSourceSet> getBuildVariantsAsGradleSourceSets(Project p
    *
    * @param project Gradle project to populate GradleSourceSet properties
    * @param variant Android Build Variant object to populate GradleSourceSet properties
+   * @param sourceSets List of source sets to which test variants will be added
+   * @param isUnitTest Indicates if the given variant is a unit test variant
    */
-  @SuppressWarnings("unchecked")
-  private static GradleSourceSet convertVariantToGradleSourceSet(Project project, Object variant) {
+  private static GradleSourceSet convertVariantToGradleSourceSet(
```

Comment:
We can change the function to return a collection of source set, and add them out of the function

Reply:
Resolved in https://github.com/microsoft/build-server-for-gradle/pull/194/commits/736a14ca181bcf75ac0c11fa65d4b691192cf3c8